llvm/test/lab2/safronov_mikhail/instrumented_test_functions.ll

```diff
@@ -0,0 +1,116 @@
+; RUN: opt -load-pass-plugin %llvmshlibdir/SafronovInstrumentFunctions%pluginext -passes=instr_func -S %s | FileCheck %s
+
+define dso_local void @instrument_start() {
+entry:
+  ret void
+}
+
+define dso_local void @instrument_end() {
+entry:
+  ret void
+}
+
+; new test.cpp
+; void empty() {
+;     return;
+; }
+
+; int mult(int a) {
+;     return a * a;
+; }
+; 
+; int increment(int x) {
+;     return x + 1;
+; }
+
+; int complexFunction(int a, int b) {
+;     return c;
+; }
+
+; struct b {};
+
+define dso_local void @_Z5emptyv() #0 {
+entry:
+  ret void
+}
+
+; CHECK-LABEL: @_Z5emptyv
+; CHECK: call void @instrument_start()
+; CHECK-NEXT: call void @instrument_end()
+; CHECK-NEXT: ret void
+
+define dso_local noundef i32 @_Z4multi(i32 noundef %a) #0 {
+entry:
```

Comment:
Could you please also add a test case with a function that already has calls to `instrument_start`/`instrument_end` functions.